packages/salesforcedx-vscode-core/src/sfdxProject/sfdxProjectConfig.ts

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2019, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { SfdxProject, SfdxProjectJson } from '@salesforce/core';
+import { AnyJson, JsonMap } from '@salesforce/ts-types';
+import * as path from 'path';
+import * as vscode from 'vscode';
+
+import { SFDX_PROJECT_FILE } from '../constants';
+import { nls } from '../messages';
+import { notificationService } from '../notifications';
+import { isSfdxProjectOpened } from '../predicates';
+import { SfdxProjectPath } from '../sfdxProject';
+import { telemetryService } from '../telemetry';
+
+/**
+ * Class representing the local sfdx-project.json file.
+ * Does not contain global values.
+ */
+export default class SfdxProjectConfig {
```

Comment:
I looked into using vscode's extension [data storage](https://code.visualstudio.com/api/extension-capabilities/common-capabilities#data-storage) capabilities to store the sfdxProjectConfig in memory, but the workspaceState, globalState, and storagePath exist on the ExtensionContext and that is only available to the `activate` function in the extensions.